crates/dekoder/src/inventory.rs

```diff
@@ -0,0 +1,84 @@
+//! Assets manager.
+use glob::glob;
+use lz4_flex::frame::FrameDecoder;
+use ndarray::Array4;
+use ndarray_npy::NpzReader;
+use std::collections::HashMap;
+use std::ffi::OsString;
+use std::fs::{read_to_string, File};
+use std::io::{BufReader, Cursor};
+use std::path::PathBuf;
+use yaml_rust2::{Yaml, YamlLoader};
+
+use crate::{EKOError, Operator, Result};
+
+/// Headers are in yaml files.
+const HEADER_EXT: &str = "*.yaml";
+
+/// Assets manager.
+pub(crate) struct Inventory<K: Eq + for<'a> TryFrom<&'a Yaml, Error = EKOError>> {
+    /// Working directory
+    pub(crate) path: PathBuf,
+    /// Available keys
+    pub(crate) keys: HashMap<OsString, K>,
+}
+
+impl<K: Eq + for<'a> TryFrom<&'a Yaml, Error = EKOError>> Inventory<K> {
+    /// Load all available entries.
+    pub fn load_keys(&mut self) -> Result<()> {
+        let path = self.path.join(HEADER_EXT);
+        let path = path
+            .to_str()
+            .ok_or(EKOError::KeyError("due to invalid path".to_owned()))?;
+        for entry in glob(path)
+            .map_err(|_| EKOError::KeyError("because failed to read glob pattern".to_owned()))?
+            .filter_map(core::result::Result::ok)
+        {
+            let cnt = YamlLoader::load_from_str(&read_to_string(&entry)?)
+                .map_err(|_| EKOError::KeyError("because failed to read yaml file.".to_owned()))?;
+            self.keys.insert(
+                entry
+                    .file_name()
+                    .ok_or(EKOError::KeyError(
+                        "because failed to read file name".to_owned(),
+                    ))?
+                    .to_os_string(),
+                K::try_from(&cnt[0])?,
+            );
+        }
+        Ok(())
+    }
+
+    /// List available keys.
+    pub fn keys(&self) -> Vec<&K> {
+        let mut ks = Vec::new();
+        for k in self.keys.values() {
+            ks.push(k);
+        }
+        ks
+    }
```

Comment:
https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.keys
(referring to `std` API, but I checked that also `hashbrown` has it, in case)

https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-FromIterator%3CT%3E-for-Vec%3CT%3E